docker/convert.go

```diff
@@ -58,20 +58,19 @@ func ConvertToAPI(serviceConfig *config.ServiceConfig, ctx project.Context, clie
 	return &result, nil
 }
 
-func isNamedVolume(volume string) bool {
-	return !strings.HasPrefix(volume, ".") && !strings.HasPrefix(volume, "/") && !strings.HasPrefix(volume, "~")
-}
-
 func volumes(c *config.ServiceConfig, ctx project.Context) map[string]struct{} {
-	volumes := make(map[string]struct{}, len(c.Volumes))
-	for k, v := range c.Volumes {
-		if len(ctx.ComposeFiles) > 0 && !isNamedVolume(v) {
-			v = ctx.ResourceLookup.ResolvePath(v, ctx.ComposeFiles[0])
+	volumes := make(map[string]struct{}, len(c.Volumes.Volumes))
+	for _, v := range c.Volumes.Volumes {
+		vol := v.Source
+		if len(ctx.ComposeFiles) > 0 && !project.IsNamedVolume(vol) {
+			vol = ctx.ResourceLookup.ResolvePath(vol, ctx.ComposeFiles[0])
+			v.Source = vol
 		}
 
-		c.Volumes[k] = v
-		if isVolume(v) {
-			volumes[v] = struct{}{}
+		// FIXME(vdemeester) why do we do that ?
```

Comment:
I should remove that, this is handled [here](https://github.com/docker/libcompose/pull/339/files#diff-33042a808eb9153e032b7eb06ed82b0eR67) 👼